lib/pure/os.nim

```diff
@@ -3193,6 +3206,30 @@ proc setLastModificationTime*(file: string, t: times.Time) {.noNimScript.} =
     discard h.closeHandle
     if res == 0'i32: raiseOSError(osLastError(), file)
 
+func isValidFilename*(filename: string, maxLen = 259.Positive): bool =
+  ## Returns true if ``filename`` is valid for crossplatform use.
+  ## This is useful if you want to copy or save files across Windows, Linux, Mac, etc.
+  ## Uses ``invalidFilenameChars``, ``invalidFilenames`` and ``maxLen`` for checking.
+  ##
+  ## .. code-block:: nim
+  ##   assert not isValidFilename(" foo")    ## Leading white space
+  ##   assert not isValidFilename("foo ")    ## Trailing white space
+  ##   assert not isValidFilename("foo.")    ## Ends with Dot
+  ##   assert not isValidFilename("con.txt") ## "CON" is invalid (Windows)
+  ##   assert not isValidFilename("OwO:UwU") ## ":" is invalid (Mac)
+  ##   assert not isValidFilename("aux.bat") ## "AUX" is invalid (Windows)
+  ##
+  # https://docs.microsoft.com/en-us/dotnet/api/system.io.pathtoolongexception
+  # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx
+  result = true
+  let f = filename.splitFile()
```

Comment:
Why the splitFile here? Isn't it the caller's job to call this with a **filename** as opposed to a full path? (Could add an assertion check that ensures this.)

Reply:
Because the recommendation on limit of length is for name+extension,
while the recommendation on naming is for name only.

Example, `"con"` is bad, but also `"con.txt"` is not recommended,
`"longname.extension"` should be less than `259` ideally, and so on.

It works with relative path and full path, and it needs to split anyway,
you want the Documentation to explicitly say that ?,
or you want an assert to force relative path only?.

Reply:
convention in os.nim is to use `path` for strings that may contain DirSep (foo/bar.txt), and filename for strings that don't (eg bar.txt)

Reply:
Done. :+1: